### ISSUE TITLE
Change SConscript() default behavoir to error if SConscript is not present.

### ISSUE THREAD

**Issue by bdbaddog** [SCons/scons#3958]
This is the completion of deprecation of old behavior.
When we introduced `must_exist` flag, the intent was to flip the default to error out and allow users to specify `must_exist=False` in the case it was ok for the file to not exist.

**Comment by mwichmann** [comment]
Should this completely clean out the warnings setup leaving just "fail unless `must_exist` is false", or leave in the warnings setup and flip the default, so that the cmdline options are still accepted but maybe don't mean anything?  (I'd need ot think about the latter)

**Comment by bdbaddog** [comment]
Ok so we added warnings in 4.2.0 and that was 07/31/2021

I'd say flip the default and remove this bit? (from SConscript.py `handle_missing_SConscript()`)

```        if SCons.Script._warn_missing_sconscript_deprecated:
            msg = (
                "Calling missing SConscript without error is deprecated.\n"
                "Transition by adding must_exist=False to SConscript calls.\n"
                "Missing SConscript '%s'" % f.get_internal_path()
            )
```

**Comment by bdbaddog** [comment]
I think that means next version must be 4.4.0

https://github.com/SCons/scons/wiki/DeprecationCycle
?

**Comment by mwichmann** [comment]
Sure, this part of the change is easy enough:
```python
diff --git a/SCons/Script/SConscript.py b/SCons/Script/SConscript.py
index 61881515e..992cf95cc 100644
--- a/SCons/Script/SConscript.py
+++ b/SCons/Script/SConscript.py
@@ -145,7 +145,7 @@ def Return(*vars, **kw):
 
 stack_bottom = '% Stack boTTom %' # hard to define a variable w/this name :)
 
-def handle_missing_SConscript(f, must_exist=None):
+def handle_missing_SConscript(f, must_exist=True):
     """Take appropriate action on missing file in SConscript() call.
 
     Print a warning or raise an exception on missing file, unless
@@ -163,22 +163,10 @@ def handle_missing_SConscript(f, must_exist=None):
           :data:`SCons.Script._no_missing_sconscript` is true.
     """
 
-    if must_exist or (SCons.Script._no_missing_sconscript and must_exist is not False):
-        msg = "Fatal: missing SConscript '%s'" % f.get_internal_path()
+    if must_exist:
+        msg = f"Fatal: missing SConscript '{f.get_internal_path()}'"
         raise SCons.Errors.UserError(msg)
 
-    if must_exist is None:
-        if SCons.Script._warn_missing_sconscript_deprecated:
-            msg = (
-                "Calling missing SConscript without error is deprecated.\n"
-                "Transition by adding must_exist=False to SConscript calls.\n"
-                "Missing SConscript '%s'" % f.get_internal_path()
-            )
-            SCons.Warnings.warn(SCons.Warnings.MissingSConscriptWarning, msg)
-            SCons.Script._warn_missing_sconscript_deprecated = False
-        else:
-            msg = "Ignoring missing SConscript '%s'" % f.get_internal_path()
-            SCons.Warnings.warn(SCons.Warnings.MissingSConscriptWarning, msg)
 
 def _SConscript(fs, *files, **kw):
     top = fs.Top
```

But leaves the question of what to do with other bits: the logic around `SCons.Script._no_missing_sconscript` - the flag setting and warning settings; and the command-line option to fiddle it.

**Comment by mwichmann** [comment]
> I think that means next version must be 4.4.0
> 
> https://github.com/SCons/scons/wiki/DeprecationCycle ?

I no longer know how to interpret that. We added the deprecation warning in 3.1.  There have been plenty of releases, and somewhat over two years, since then, what constitutes major now?  e.g. Python now considers their annual realase to be basically major, while 2. to 3. wa something different, perhaps a "generational" change.  One reading says it would need to be 5.0, which I hope isn't the case.  The earlier things I kicked out were deprecated in or before the 1.0 timeframe so really no questions there.
